src/libm/sleefsimdsp.c

```diff
@@ -1678,11 +1678,19 @@ EXPORT CONST vfloat xfabsf(vfloat x) { return vabs_vf_vf(x); }
 EXPORT CONST vfloat xcopysignf(vfloat x, vfloat y) { return vcopysign_vf_vf_vf(x, y); }
 
 EXPORT CONST vfloat xfmaxf(vfloat x, vfloat y) {
+#if SLEEF_SINGLE_MINMAXNUM_AVAILABLE
```

Comment:
Hello @blu , thank you for working on this. This is a good optimization.
My preference though would be to avoid the preprocessor code in the `sleefsimd[s|d]p.c` and just define the `vmaxnum_vf_vf_vf` and `vminnum_vf_vf_vf` (and their double precision counterparts) in the helper files.

To avoid duplicating the same definition in all the helper files that do not have the underlying direct instruction, would be to write the generic methods in `df.h` and `dd.h`, `#ifndef` guarded with the macros you created, and define those macros in the helper files that have such instructions (AdvSIMD at the moment).

This should minimize the amount of code written for this optimization, especially if we find other places in the library where we can use it.